src/Symfony/Component/HttpClient/RetryableHttpClient.php

```diff
@@ -168,4 +190,14 @@ private function passthru(AsyncContext $context, ?ChunkInterface $firstChunk, st
 
         yield $lastChunk;
     }
+
+    private static function shiftBaseUri(array $options, array &$baseUris): array
+    {
+        if ($baseUris) {
+            $baseUri = 1 < \count($baseUris) ? array_shift($baseUris) : current($baseUris);
+            $options['base_uri'] = \is_array($baseUri) ? $baseUri[array_rand($baseUri)] : $baseUri;
```

Comment:
@nicolas-grekas I just noticed something odd here, but maybe it's intentional?
The provided array to be shuffled is ditched after only one randomized URI is extracted. Is it normal or should I make a fix to use `current` instead of `array_shift` for nested arrays?

Reply:
We have two different expectation apparently: mine is to pick one randomly. Yours is to shuffle and flatten.
The doc PR tells about my way. I don't know how to decide between both approaches honeslty.

Reply:
That's a tough one indeed. I honestly don't have enough use cases on my own to decide if it's truly needed to try all the URIs from the array or just one.
Maybe @lyrixx you have an opinion, since the base issue was yours? (even if the shuffling part was not in the original need)

Reply:
I'm 👍🏼 with @nicolas-grekas

Reply:
Thanks for the opinion! Leaving it like that then.